qaoa_training_pipeline/training/scipy_trainer.py

```
@@ -149,10 +159,19 @@ def from_config(cls, config: dict) -> "ScipyTrainer":
         """Create a scipy trainer based on a config."""
 
         evaluator_cls = EVALUATORS[config["evaluator"]]
-        evaluator = evaluator_cls.from_config(config["evaluator_init"])
-        minimize_args = config["minimize_args"]
 
-        return cls(evaluator, minimize_args)
+        if "qaoa_angles_function" not in config:
+            function = IdentityFunction()
+        else:
+            function_cls = FUNCTIONS[config["qaoa_angles_function"]]
+            function = function_cls.from_config(config["qaoa_angles_function_init"])
```

Shall we raise an error if the `config["qaoa_angles_function"]` is not a supported function?